build.rs

```diff
@@ -236,6 +291,12 @@ fn get_r_version_strings(r_paths: &InstallationPaths) -> io::Result<RVersionInfo
     })
 }
 
+fn get_r_version(r_version_env_var: &str, r_paths: &InstallationPaths) -> io::Result<RVersionInfo> {
+    get_r_version_from_env(r_version_env_var)
+    .ok_or(Error::new(ErrorKind::Other, "Cannot determine R version from environement variable"))
+    .or_else(|_| get_r_version_from_r(r_paths))
```

Comment:
I think the logic here should be

1. if `R_VERSION` exists and it's in the expected format, use it.
2. if `R_VERSION` exists but it's not in the expected format, return an error
3. if `R_VERSION` doesn't exist, and if the execution of `R` returns the output in the expected format, use it
4. if `R_VERSION` doesn't exist, and if the execution of `R` returns the output in the expected format (or the execution failed), return an error

I mean, the malformed `R_VERSION` should not be ignored by `or_else()`. What do you think?

Reply:
Also sounds good to me. In this example the argument of `ok_or` is actually never observed. I can modify code to return something more meaningful for the case of  env var.

As an alternative, we can leave it as is but add a warning to the output stating that `R_VERSION` is malformed and therefore skipped.

Reply:
> As an alternative, we can leave it as is but add a warning to the output stating that `R_VERSION` is malformed and therefore skipped.

I was thinking which would be better. Are there chances that `R_VERSION` environmental variable is unintentionally set (e.g. some other application use the same name of the envvar)? If not, I think it should stop with error. The case when the user provide `R_VERSION` is probably when the auto-detection won't work correctly, so it should not fall back to the method.

Reply:
I am sure it is possible that `R_VERSION` is used by someone/some toolchain -- I for instance use it locally on my machines to quickly switch environment between different R installations. But I think it can be properly set/unset before compilation if needed. I have modified the build script so now if `R_VERSION` is present but invalid, it fails, but if `R_VERSION` is missing completely, it falls back to querying R.